flink-core/src/test/java/org/apache/flink/core/memory/HybridOnHeapMemorySegmentTest.java

```diff
@@ -77,4 +77,32 @@ public void testHybridHeapSegmentSpecifics() {
 		assertEquals(3, buf2.position());
 		assertEquals(7, buf2.limit());
 	}
+
+	@Test
+	public void testReadOnlyByteBufferPut() {
+		final byte[] buffer = new byte[100];
+		HybridMemorySegment seg = new HybridMemorySegment(buffer);
+
+		String content = "hello world";
+		ByteBuffer bb = ByteBuffer.allocate(20);
+		bb.put(content.getBytes());
+		bb.rewind();
+
+		int offset = 10;
+		int numBytes = 5;
+		seg.put(offset, bb.asReadOnlyBuffer(), numBytes);
```

Comment:
Are we certain that `bb.asReadOnlyBuffer()` returns a buffer that is neither direct, nor has an array? Only on that case would it go into the affected code branch.

The test should guard its assumption like:
```java
ByteBuffer buf = bb.asReadOnlyBuffer();
assertFalse(buf.isDirect());
assertFalse(buf.hasArray());
```

Reply:
Yes. If a heap byte buffer is read only, its hasArray will be false. I have added the assertions to the UT. Thanks for the advice.